hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java

```diff
@@ -0,0 +1,603 @@
+	@SuppressWarnings("deprecation")
+	public boolean useMaxForLimit() {
+		// Does the LIMIT clause take a "maximum" row number instead of a total number of returned rows?
+		return true;
+	}
+
+	@Override
+	@SuppressWarnings("deprecation")
+	public String getLimitString(String sql, boolean hasOffset) {
+		if ( hasOffset ) {
+			throw new UnsupportedOperationException( "query result offset is not supported" );
+		}
+
+		// This does not support the InterSystems IRIS SQL 'DISTINCT BY (comma-list)' extensions,
+		// but this extension is not supported through Hibernate anyway.
+		final int insertionPoint = sql.startsWith( "select distinct" ) ? 15 : 6;
+
+		return new StringBuilder( sql.length() + 8 )
+				.append( sql )
+				.insert( insertionPoint, " TOP ? " )
+				.toString();
+	}
+
+	// callable statement support ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+	@Override
+	public int registerResultSetOutParameter(CallableStatement statement, int col) throws SQLException {
+		return col;
+	}
+
+	@Override
+	public ResultSet getResultSet(CallableStatement ps) throws SQLException {
+		ps.execute();
+		return (ResultSet) ps.getObject( 1 );
+	}
+
+	// miscellaneous support ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+	@Override
+	public String getLowercaseFunction() {
+		// The name of the SQL function that transforms a string to lowercase
+		return "lower";
+	}
+
+	@Override
+	public String getNullColumnString() {
+		return "";
+	}
+
+	@Override
+	public JoinFragment createOuterJoinFragment() {
+		// Create an OuterJoinGenerator for this dialect.
+		return new InterSystemsIRISJoinFragment();
+	}
+
+	@Override
+	public String getNoColumnsInsertString() {
+		// The keyword used to insert a row without specifying
+		// any column values
+		return " default values";
+	}
+
+	@Override
+	public SQLExceptionConversionDelegate buildSQLExceptionConversionDelegate() {
+		return new InterSystemsIRISSQLExceptionConversionDelegate( this );
+	}
+
+	@Override
+	public ViolatedConstraintNameExtracter getViolatedConstraintNameExtracter() {
+		return EXTRACTER;
+	}
+
+	/**
+	 * The InterSystemsI RIS ViolatedConstraintNameExtracter.
+	 */
+	public static final ViolatedConstraintNameExtracter EXTRACTER = new TemplatedViolatedConstraintNameExtracter() {
+		@Override
+		protected String doExtractConstraintName(SQLException sqle) throws NumberFormatException {
+			return extractUsingTemplate( "constraint (", ") violated", sqle.getMessage() );
+		}
+	};
+
+
+	// Overridden informational metadata ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+	@Override
+	public boolean supportsEmptyInList() {
+		return false;
+	}
+
+	@Override
+	public boolean areStringComparisonsCaseInsensitive() {
+		return true;
+	}
+
+	@Override
+	public boolean supportsResultSetPositionQueryMethodsOnForwardOnlyCursor() {
+		return false;
+	}
+
+	/**
+	 * ddl like ""value" integer null check ("value">=2 AND "value"<=10)" isn't supported
+	 */
+	@Override
+	public boolean supportsColumnCheck()
+	{
```

Comment:
Begin braces, `{` should begin on the line where the method is defined, not on a new-line.
There are several occurrences of this in this file.